nipype/algorithms/mesh.py

```diff
@@ -13,55 +13,78 @@
 
 
 import numpy as np
-from scipy.spatial.distance import euclidean
+from numpy.linalg import norm
+import os.path as op
+from nipype.external import six
 
 from .. import logging
 
 from ..interfaces.base import (BaseInterface, traits, TraitedSpec, File,
                                BaseInterfaceInputSpec)
+from warnings import warn
 iflogger = logging.getLogger('interface')
 
 
-class P2PDistanceInputSpec(BaseInterfaceInputSpec):
+class ComputeMeshWarpInputSpec(BaseInterfaceInputSpec):
     surface1 = File(exists=True, mandatory=True,
-                    desc=("Reference surface (vtk format) to which compute "
-                          "distance."))
+                    desc=('Reference surface (vtk format) to which compute '
+                          'distance.'))
     surface2 = File(exists=True, mandatory=True,
-                    desc=("Test surface (vtk format) from which compute "
-                          "distance."))
-    weighting = traits.Enum("none", "surface", usedefault=True,
-                            desc=('"none": no weighting is performed, '
-                                  '"surface": edge distance is weighted by the '
-                                  'corresponding surface area'))
+                    desc=('Test surface (vtk format) from which compute '
+                          'distance.'))
+    metric = traits.Enum('euclidean', 'sqeuclidean', usedefault=True,
+                         desc=('norm used to report distance'))
+    weighting = traits.Enum(
+        'none', 'area', usedefault=True,
+        desc=('"none": no weighting is performed, surface": edge distance is '
+              'weighted by the corresponding surface area'))
+    out_warp = File('surfwarp.vtk', usedefault=True,
+                    desc='vtk file based on surface1 and warpings mapping it '
+                    'to surface2')
+    out_file = File('distance.npy', usedefault=True,
+                    desc='numpy file keeping computed distances and weights')
 
 
-class P2PDistanceOutputSpec(TraitedSpec):
+class ComputeMeshWarpOutputSpec(TraitedSpec):
     distance = traits.Float(desc="computed distance")
+    out_warp = File(exists=True, desc=('vtk file with the vertex-wise '
+                                       'mapping of surface1 to surface2'))
+    out_file = File(exists=True,
+                    desc='numpy file keeping computed distances and weights')
 
 
-class P2PDistance(BaseInterface):
+class ComputeMeshWarp(BaseInterface):
 
-    """Calculates a point-to-point (p2p) distance between two corresponding
-    VTK-readable meshes or contours.
+    """
+    Calculates a the vertex-wise warping to get surface2 from surface1.
+    It also reports the average distance of vertices, using the norm specified
+    as input.
+
+    .. warning:
+
+      A point-to-point correspondence between surfaces is required
```

Comment:
what if these two files are spheres? is there a way to weight them differently, for example with curvature information or some such?